src/inmanta/data/__init__.py

```diff
@@ -3064,8 +3054,8 @@ async def persist_on_resume(cls, env: uuid.UUID, connection: Optional[asyncpg.co
         Restores default halted state. Returns a list of agents that should be unpaused.
         """
 
-        async def query_with_connection(connection: asyncpg.connection.Connection) -> List[str]:
-            async with connection.transaction():
+        async with cls.get_connection(connection) as con:
+            async with con.transaction():
```

Comment:
We should use `con` instead of `connection` in the transaction block.

Reply:
Oh, thanks, good catch.

Reply:
Looks like I already fixed this yesterday.

Reply:
You didn't push this yet, I guess?

Reply:
Oh, I misunderstood, I fixed the `connection.transaction()` -> `con.transaction()` but I didn't fix the body... I fixed it now.